### PR TITLE
Add support for more than 65535 elf segments

### LINKED ISSUE #1485: Support for more than 65535 segments in ELF64
Filing it here, to not forget. From https://github.com/radareorg/radare2/issues/18783#issue-907516761

```
e_phnum
This member holds the number of entries in the program header table.  Thus the product of e_phentsize and e_phnum gives the table's size in bytes.  If a file has no program header, e_phnum holds the value zero.
If the number of entries in the program header table is larger than or equal to PN_XNUM (0xffff), this member holds PN_XNUM (0xffff) and the real number of entries in the program header table is  held  in  the  sh_info member of the initial entry in section header table.  Otherwise, the sh_info member of the initial entry contains the value zero.

PN_XNUM
This is defined as 0xffff, the largest number e_phnum can have, specifying where the actual number of program headers is assigned.
```

### PR DESCRIPTION
**Pull request by ghost**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Add support for more than 65535 using section information.

**Test plan**

I don't know If I can upload the test binary (needed to add a test).

```
rizin test_elf
> iSS~?
```

**Closing issues**

close https://github.com/rizinorg/rizin/issues/1485

### CODE REVIEW

File: librz/bin/format/elf/elf_segments.c [hunk 1]
```diff
@@ -98,6 +98,25 @@ static RzVector *get_segments_from_phdr(ELFOBJ *bin, RzBinObjectLoadOptions *opt
 	return result;
 }
 
+static size_t get_number_of_segments(ELFOBJ *bin, RzVector *sections) {
+	if (bin->ehdr.e_phnum != PN_XNUM) {
+		return bin->ehdr.e_phnum;
+	}
+
+	if (!sections) {
+		RZ_LOG_WARN("Failed to fetch the number of segments because there is no sections.\n");
```
Comment (ret2libc) [inline librz/bin/format/elf/elf_segments.c]:
```suggestion
		RZ_LOG_WARN("Failed to fetch the number of segments because there are no sections.\n");
```

**Comment by ret2libc** [comment]
Could you paste in the description the old output behaviour and the new one?

**Comment by ghost** [comment]
old
```
> rizin test_elf
Cannot retrieve regstate on: AMD x86-64 architecture (not yet supported)
Setting up coredump arch-bits to: x86-64
 -- Print the contents of the current block with the 'p' command
[0x00001000]> iSS~?
65539
```

new
```
> rizin test_elf
 ~/Downloads -> rizin test_elf
Cannot retrieve regstate on: AMD x86-64 architecture (not yet supported)
Setting up coredump arch-bits to: x86-64
 -- If you want to open the file in read-write mode, invoke rizin with '-w'
[0x00001000]> iSS~?
131074
```

**Comment by ghost** [comment]
Should I upload the test binary that come from the radare2 issue?

**Comment by ret2libc** [comment]
I think you can, yes. @XVilka cc. If it's in radare2-testbins it means it can be shared IMO.

**Review by ret2libc — APPROVED** [review]

**Comment by ghost** [comment]
After the `rizin-testbin` PR will be merged, it should be green.

**Comment by ghost** [comment]
I have restarted the CI.

**Review by ret2libc — APPROVED** [review]

**Pull request merged**